### ISSUE TITLE
Direct linking

### ISSUE THREAD

**Issue by tbeu** [ralight/mallocfail#1]
Is there an description to directly link to mallocfail instead of the preload hook?

**Comment by ralight** [comment]
No, I'm not sure whether that is practical. To do that you'd have to replace all of libc wouldn't you?

**Comment by tbeu** [comment]
Issue is, that I constantly fail to run mallocfail pre-hooked executables through Travis CI. See https://travis-ci.org/tbeu/matio/builds and check branch mallocfail.

**Comment by ralight** [comment]
Looking at this one: https://travis-ci.org/tbeu/matio/jobs/448055273

It seems as though it's working as expected, printing out the stack traces when it fails them, there are errors codes returned by your code when memory allocations fail and even a crash.

What else would you expect to see?
